components/script/dom/websocket.rs

```diff
@@ -333,6 +333,31 @@ impl WebSocketMethods for WebSocket {
 
     // https://html.spec.whatwg.org/multipage/#dom-websocket-send
     fn Send(&self, data: USVString) -> Fallible<()> {
+
+        let bufferedAmount = self.buffered_amount.get() + (data.0.as_bytes().len() as u32)
+
+        self.Send_Impl(bufferedAmount)
+
+        let _ = my_sender.lock().unwrap().send_message(Message::Text(data.0));
+
+        Ok(())
+    }
+
+    // https://html.spec.whatwg.org/multipage/#dom-websocket-send
+    fn Send(&self, data: Blob) -> Fallible<()> {
+
+        let bufferedAmount = self.buffered_amount.get() + (data.size as u32)
+
+        self.Send_Impl(bufferedAmount)
+
+        let _ = data.send(my_sender.lock().unwrap().send_message)
```

Comment:
The way http://mxr.mozilla.org/servo/source/components/script/dom/filereader.rs#357 works is it creates a channel (ie. a pair of Sender/Receiver values) that are used to extract the blob's contents. The blob sends its contents using the Sender, and whenever the consumer wants to get the actual value, it uses the receiver (see http://mxr.mozilla.org/servo/source/components/script/dom/filereader.rs#418). I recommend calling recv immediately here.